### DISCUSSION TITLE
Read MJPG or RTSP stream from IP camera and pass it to ZXing.net to read QR-code

### DISCUSSION THREAD

**Discussion by MichaelUray** [micjahn/ZXing.Net#314] [Q&A] [answered]
Hi guys,

I am wondering what would be the best way to read a live stream from a Dahua IP camera (DH-TPC-DF1241-HTM) to decode a QR-code. The camera supports to to stream the video via RTSP (H.264) or MJPG.

There is a lack of knowledge about the .NET framework and its streams on my side, but I have some experience in C programming and LabVIEW where I actually want to implement it.

I was able to get a MJPG stream via a .NET httpclient in LabVIEW (ReadAsync provides JPG data), but I am not sure how to process it to the ZXing.net library.
The LabVIEW program stucks at the moment when I pass the MJPG stream to System.Drawing.Bitmap, maybe is this the wrong way.

I know the LabVIEW programming language is not very common, but I attach there a screenshot and you might be able to understand it since it is a graphical programming language.
![2021-03-07_11-40-22_test_http_client_2 vi_Block_Diagram_](https://user-images.githubusercontent.com/25169478/110237119-4f49cf00-7f3a-11eb-8030-ebf374b4181b.png)

Any idea what I do wrong there, or is there any better way to read a QR code via a network camera?

Michael

**Comment by micjahn** [comment, marked as answer]
I doesn't have any experience with LabVIEW but in my opinion System.Drawing.Bitmap doesn't support MJPG. I think you have to convert the MJPG stream to single JPG images and then you can use them with the Bitmap class.

**Reply by MichaelUray** [reply]
Thanks, to pass a single JPG image to the ZXing library worked for me.
